backend/tests/unit/core/test_annotation_unit.py

```diff
@@ -10,13 +11,60 @@ def test_annotation_unit_gets_missing_dependencies(annotation_unit_lmna):
     assert actual == ['HGNC_ID']
 
 
+def test_annotation_unit_ready_for_annotation(annotation_unit_lmna):
+    """Verifies if the annotation unit is ready for annotation"""
+
+    missing = annotation_unit_lmna.get_missing_dependencies()
+    missing_dependency = missing[0]
+    mock_genomic_unit_collection = Mock()
+    mock_genomic_unit_collection.find_genomic_unit_annotation_value = Mock()
+    dependency_annotation = mock_genomic_unit_collection.find_genomic_unit_annotation_value(
+        annotation_unit_lmna.genomic_unit, missing_dependency
+    )
+
+    actual = annotation_unit_lmna.ready_for_annotation(dependency_annotation, missing_dependency)
+    assert actual is True
+
+    dependency_annotation = ""
+    actual = annotation_unit_lmna.ready_for_annotation(dependency_annotation, missing_dependency)
+    assert actual is False
+
+
+def test_annotation_unit_should_continue_annotation(annotation_unit_lmna, annotation_unit_lmna_exceeded_delay_count):
+    """
+    Verifies if the annotation unit should continue annotation for both cases
+    of annotation_unit with delay_count not exceeded and exceeded.
+    """
+
+    (actual_missing_dependencies, actual_logger_message) = annotation_unit_lmna.should_continue_annotation()
+    assert actual_missing_dependencies == []
+    assert actual_logger_message == '%s Delaying Annotation, Missing Dependency...'
+
+    (actual_missing_dependencies,
```

Comment:
Thanks for adding this test case to the unit tests!  Something that will get lost when running this unit test is that it will only count one unit test to be executed in the test report.  It is easier to track which tests are failing and why if this was extracted to be in a separate unit test function.

So this unit test name would become `test_annotation_unit_should_continue_annotation_delays` and the additional unit test case function would be `test_annotation_unit_should_not_continue_annotation_canceling`.

Reply:
Looks great!